worker/rhpv3.go

```diff
@@ -52,20 +52,62 @@ var (
 	errBalanceMaxExceeded = errors.New("ephemeral account maximum balance exceeded")
 )
 
-func (w *worker) FetchRevisionWithAccount(ctx context.Context, pt rhpv3.HostPriceTable, hostKey types.PublicKey, siamuxAddr string, bh uint64, contractID types.FileContractID) (rev types.FileContractRevision, err error) {
-	if breakdown := GougingCheckerFromContext(ctx).Check(nil, &pt); breakdown.Gouging() {
-		return types.FileContractRevision{}, fmt.Errorf("failed to fetch revision, %w: %v", errGougingHost, breakdown.Reasons())
+// FetchRevision fetches the latest revision of a contract and uses an account
+// as the primary payment method. If the account balance is insufficient, it
+// falls back to using the contract as a payment method.
+func (w *worker) FetchRevision(ctx context.Context, timeout time.Duration, contract api.ContractMetadata, bh uint64, lockPriority int, lockDuration time.Duration) (types.FileContractRevision, error) {
+	timeoutCtx := func() (context.Context, context.CancelFunc) {
+		if timeout > 0 {
+			return context.WithTimeout(ctx, timeout)
+		}
+		return ctx, func() {}
+	}
+	if timeout != 0 && timeout < lockDuration {
```

Comment:
I think this should probably error out? I don't see why the `lockDuration` should be set to the `timeout` they are two different things. In the case of active contracts for instance I think the `lockDuration` should be 0 because we only want to fetch the revision and nothing more.